File: app/src/main/java/com/example/cxh/kotlinstudy/kotlin/kt09_singleton/java/PlainOldSingleton.java
package com.example.cxh.kotlinstudy.kotlin.kt09_singleton.java;

// 饿汉式
public class PlainOldSingleton {
    private static PlainOldSingleton INSTANCE = new PlainOldSingleton();

    private PlainOldSingleton(){
        System.out.println("PlainOldSingleton");
    }

    public static PlainOldSingleton getInstance(){
        return INSTANCE;
    }
}

File: app/src/main/java/com/example/cxh/kotlinstudy/kotlin/kt09_singleton/kotlin/PlainOldSingleton.kt
package com.example.cxh.kotlinstudy.kotlin.kt09_singleton.kotlin

// 饿汉式
class PlainOldSingleton private constructor() {

    init {
        println("PlainOldSingleton")
    }

    companion object {
        val instance = PlainOldSingleton()
    }
}
